ivy/functional/ivy/creation.py

```diff
@@ -1100,6 +1100,158 @@ def eye(
     )
 
 
+@to_native_arrays_and_back
+@handle_out_argument
+@infer_dtype
+@infer_device
+@handle_nestable
+@handle_exceptions
+@handle_array_like_without_promotion
+def eye_like(
+    x: Union[ivy.Array, ivy.NativeArray],
+    /,
+    *,
+    k: int = 0,
+    dtype: Optional[Union[ivy.Dtype, ivy.NativeDtype]] = None,
+    device: Optional[Union[ivy.Device, ivy.NativeDevice]] = None,
+    out: Optional[ivy.Array] = None,
+) -> ivy.Array:
+    """Returns a array filled with ones on the k diagonal and zeros elsewhere. having
+    the same ``shape`` as an input array ``x``.
+
+
+    Parameters
+    ----------
+    x
+         input array from which to derive the output array shape.
+    k
+        index of the diagonal. A positive value refers to an upper diagonal, a negative
+        value to a lower diagonal, and 0 to the main diagonal. Default: ``0``.
+    dtype
+        output array data type. If dtype is None, the output array data type must be the
+        default floating-point data type. Default: ``None``.
+    device
+        the device on which to place the created array.
+    out
+        optional output array, for writing the result to. It must have a shape that the
+        inputs broadcast to.
+
+    Returns
+    -------
+    ret
+        an array having the same shape as ``x`` and filled with ``ones`` in diagonal ``k``
+        and ``zeros`` elsewhere.
+
+    This function conforms to the `Array API Standard
+    <https://data-apis.org/array-api/latest/>`_. This docstring is an extension of the
+    `docstring <https://data-apis.org/array-api/latest/API_specification/generated/signatures.creation_functions.eye.html>`_ # noqa
+    in the standard.
+
+    Both the description and the type hints above assumes an array input for simplicity,
+    but this func;tion is *nestable*, and therefore also accepts :class:`ivy.Container`
+    instances as a replacement to any of the arguments.
+
+    Functional Examples
+    -------------------
+
+    With :class:`ivy.Array` input:
+
+    >>> x1 = ivy.array([0, 1],[2, 3])
+    >>> y1 = ivy.eye_like(x1)
+    >>> print(y1)
+    ivy.array([[1., 0.],
+               [0., 1.]])
+
+    >>> x1 = ivy.array([0, 1, 2],[3, 4, 5],[6, 7, 8])
+    >>> y1 = ivy.eye_like(x1)
+    >>> print(y1)
+    ivy.array([[1., 0., 0.],
+               [0., 1., 0.],
+               [0., 0., 1.]])
+
+    With :'k' input:
```

Comment:
You shouldn't mention `k` as input example alone you can only use `ivy` instances means you can either mix it with `ivy.Array` examples

Reply:
okay, thanks a lot for the clarification